keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -333,3 +340,36 @@ def _ensure_inputs_are_compute_dtype(self, inputs):
                 self.compute_dtype,
             )
         return inputs
+
+    def _pad_bounding_boxes(self, bounding_boxes):
+        if not isinstance(bounding_boxes, tf.RaggedTensor):
+            bounding_boxes = tf.RaggedTensor.from_tensor(bounding_boxes)
+        pad_value = tf.cast(-1, dtype=self.compute_dtype)
+        padded_bounding_box = bounding_boxes.to_tensor(pad_value)
+        return padded_bounding_box
+
+    def _filter_padded_boxes(self, bounding_boxes):
+        def is_padded(box):
```

Comment:
this is_padded check seems unstable; we should check the class ID and see if it is -1 instead of relying on a logical and like this